tgui/packages/tgui/interfaces/common/sorting/SortableTable/utils.ts

```diff
@@ -0,0 +1,72 @@
+/**
+ * @file
+ * @copyright 2024
+ * @author mloccy (https://github.com/mloccy)
+ * @license ISC
+ */
+
+import { SortDirection } from "../type";
+import { SearchState, SortableTableHeaderConfig, SortableTableRowProps, SortState } from "./type";
+
+export const onSortClick = (index: number, current: SortState, setCallback: (nextState: SortState) => void) => {
+  if (current !== null) {
+    if (current.index === index) {
+      setCallback({
+        dir: (current.dir === SortDirection.Asc ? SortDirection.Desc : SortDirection.Asc),
+        index: index,
+      });
+    } else {
+      setCallback({ dir: SortDirection.Asc, index: index });
+    }
+  } else {
+    setCallback({ dir: SortDirection.Asc, index: index });
+  }
+};
+
+export const sortAndFilterRows = (
+  headerConfig: SortableTableHeaderConfig[],
+  rowConfig: SortableTableRowProps[],
+  sortState: SortState,
+  searchState: SearchState): SortableTableRowProps[] => {
+
+  const first = rowConfig[0];
+  let sorted = rowConfig;
+  if (headerConfig.length !== first.cells.length) {
+    throw new Error(`Inconsistent number of fields in table: ${headerConfig.length}x${first.cells.length}`);
+  }
+
+  let header: SortableTableHeaderConfig;
+
+  if (sortState) {
+    header = headerConfig[sortState.index];
+  } else if (searchState) {
+    header = headerConfig[searchState.index];
+  } else {
+    return rowConfig;
+  }
```

Comment:
Probably easiest just to have an early return, then handle the sort and search as you are below? (i.e. one after the other?) Pre-loading the `header` here doesn't seem especially useful.

Something like:

```
if (!sortState && !searchState) {
  return;
}
```